### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by matiasgarcia91**

### CODE REVIEW

File: src/resolvers/Mutation.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+function postBudget(parent, args, context) {
+  const userId = getUserId(context);
+  return context.prisma.createBudget({
+    total: args.total,
+    startDate: args.startDate,
+    endDate: args.endDate,
+    savingsTarget: args.savingsTarget,
+    postedBy: { connect: { id: userId } }
```
Comment (matiasgarcia91) [inline src/resolvers/Mutation.js:40]: this is just so you know it's possible. You can destructure arguments on the function definition like:

```
function postBudget(parent, { total, startData, endDate, savingsTarget }, context) {
```

File: src/resolvers/Mutation.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      },
+      budget: {
+        id: args.budgetId
+      }
+    }
+  });
+  const totalExpenses = allExpenses.reduce((acc, curr) => {
+    if (curr.expenseAmount) {
+      return (acc = acc + curr.expenseAmount);
+    }
```
Comment (matiasgarcia91) [inline src/resolvers/Mutation.js:69]: you don't need to assign acc again, you can just return `acc + curr.expenseAmount`. Also you should `return acc` outside of that `if` so if anything happens and one of the expenses doesn't have an `expenseAmount` you still don't lose the total

File: src/resolvers/Mutation.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+  }
+  const budgets = await context.prisma.budgets({
+    where: {
+      postedBy: {
+        id: userId
+      }
+    },
+    orderBy: "endDate_DESC"
+  });
+  budgets.map(async budget => {
```
Comment (matiasgarcia91) [inline src/resolvers/Mutation.js:119]: this is a nice one. Whenever we use array methods we prefer that they don't have "side-effects". These can be for example making calls to the DB for creation of stuff. Map method is to transform one array into another and we should keep it that way so you have two options:
- You keep using map but you map the array into an array of promises and then use Promise.all to await on them all, which makes it evident that the new array is going to perform other stuff outside of just transforming one value into another.
- You switch to using `.forEach` which reads better for this as "for each of these do this"

**Review by matiasgarcia91 — COMMENTED** [review]
Really clean code Shashank! Congrats keep it up

File: src/resolvers/Mutation.js (hunk 3 above)
Comment (matiasgarcia91) [reply src/resolvers/Mutation.js:119]: this is a nice reference: https://medium.com/@ntgard/foreach-is-for-side-effects-60fca9f78850

File: src/resolvers/Mutation.js (hunk 1 above)
Comment (sneelagiri) [reply src/resolvers/Mutation.js:40]: Looks better this way

File: src/resolvers/Mutation.js (hunk 2 above)
Comment (sneelagiri) [reply src/resolvers/Mutation.js:69]: I see. I will do that.

File: src/resolvers/Mutation.js (hunk 3 above)
Comment (sneelagiri) [reply src/resolvers/Mutation.js:119]: I will start using for each

**Pull request merged**